File: base_layout/src/main0/java/com/example/t00/ViewPagerAdapter.java
package com.example.t00;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.ArrayList;

public class ViewPagerAdapter extends FragmentPagerAdapter {

    private final ArrayList<Fragment> mFragmentList = new ArrayList<>();

    public ViewPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {
        // 这里可以根据需要添加条件逻辑
        if (position == 1) {
            // 处理 position 为 1 的情况（如果需要）
        }
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    public void addFragment(Fragment fragment) {
        mFragmentList.add(fragment);
        notifyDataSetChanged();
    }

    public void addFragmentAt(Fragment fragment, int position) {
        mFragmentList.add(position, fragment);
        notifyDataSetChanged();
    }

    private static final String TAG = "ViewPagerAdapter";
}

File: base_layout/src/main/java/com/cloudsification/clouds/ViewPagerAdapter.kt
package com.cloudsification.clouds

import java.util.*

/**
 * Created by Depression on 10-08-2018.  需要一个父类的fragment manager 直接support的那个
 */
class ViewPagerAdapter(fm: androidx.fragment.app.FragmentManager) : androidx.fragment.app.FragmentPagerAdapter(fm) {

    private val mFragmentList = ArrayList<androidx.fragment.app.Fragment>()

    override fun getItem(position: Int): androidx.fragment.app.Fragment {                   // 翻页的时候一直调用的这个
        if(position == 1){

        }
        return mFragmentList[position]
    }

    override fun getCount(): Int {
        return mFragmentList.size
    }

    fun addFragment(fragment: androidx.fragment.app.Fragment) {
        mFragmentList.add(fragment)
        notifyDataSetChanged()
    }

    fun addFragmentAt(fragment: androidx.fragment.app.Fragment, position: Int) {
        mFragmentList.add(position, fragment)
        notifyDataSetChanged()
    }

    companion object {
        private val TAG = "ViewPagerAdapter"
    }
}


// 滑动到新的界面的销毁问题 以及新的fragment的初始化问题